compiler/executable/match_/planner/plan.rs

```diff
@@ -1676,7 +1653,7 @@ impl ConjunctionPlan<'_> {
                 };
 
                 let direction = if matches!(inputs, Inputs::None([])) {
-                    let CostMetaData::Direction(unbound_direction) = metadata else {
+                    let CostMetaData::Direction(unbound_direction, _) = metadata else {
```

Comment:
We could verify here that the metadata sort var agrees with the input `sort_variable`.

Reply:
Might as well, if you think "join of size one" isn't confusing. I'd say it's not.